test/fuzz/lua_fuzzer/serializer.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include "lua_grammar.pb.h"
+
+using namespace lua_grammar;
+
+#define PROTO_TOSTRING(PROTO_TYPE, VAR_NAME) \
+	std::string PROTO_TYPE##ToString(const PROTO_TYPE & (VAR_NAME))
+
+/* PROTO_TOSTRING version for nested (depth=2) protobuf messages. */
+#define NESTED_PROTO_TOSTRING(PROTO_TYPE, VAR_NAME, PARENT_MESSAGE) \
+	std::string PROTO_TYPE##ToString \
+	(const PARENT_MESSAGE::PROTO_TYPE & (VAR_NAME))
```

Comment:
These macros obfuscate the code. I don't understand how this

```CXX
PROTO_TOSTRING(Block, block);
NESTED_PROTO_TOSTRING(ReturnOptionalExpressionList, explist, LastStatement);
```

is better than this

```CXX
std::string BlockToString(const Block &block);
std::string ReturnOptionalExpressionListToString(
        const LastStatement::ReturnOptionalExpressionList &exprlist);
```

Looking at the latter code snippet I see that these are two function declarations. Looking at the former, I can't figure out what it is without looking at the macros. Generally, we discourage using macros like this in our code.

Reply:
> These macroses provided by LPM for better readability.

But these macros aren't third-party - they are defined in this patch. And they definitely don't improve readability, quite the contrary.

Reply:
This is code with macros:
```c
NESTED_PROTO_TOSTRING(ReturnOptionalExpressionList, explist, LastStatement);
```

And without:
```c
std::string
ReturnOptionalExpressionListToString(const LastStatement::ReturnOptionalExpressionList &LastStatement);
```
@locker isn't macro usage better for readability?

Reply:
No, it isn't better IMO. The macro obfuscates the code making it more difficult to understand.

Reply:
I don't like the macro too. The problem is that you need to know what exactly the macro does to understand what happens here. The code is more simple to read if you need to know less to understand in.
BTW is to possible to simply write:
```
std::string
ToString(const LastStatement::ReturnOptionalExpressionList &obj);
```